karapace/schema_registry_apis.py

```diff
@@ -241,8 +227,27 @@ def send_kafka_message(self, key, value):
         future = self.producer.send(self.config["topic_name"], key=key, value=value)
         self.producer.flush(timeout=self.kafka_timeout)
         msg = future.get(self.kafka_timeout)
-        self.log.debug("Sent kafka msg key: %r, value: %r, offset: %r", key, value, msg.offset)
-        self.get_offset_from_queue(msg.offset)
+        sent_offset = msg.offset
+
+        self.log.info(
+            "Waiting for schema reader to caugth up. key: %r, value: %r, offset: %r",
+            key,
+            value,
+            sent_offset,
+        )
+
+        if self.ksr.offset_watcher.wait_for_offset(sent_offset, timeout=1 * MINUTE) is True:
+            self.log.info(
+                "Schema reader has found key. key: %r, value: %r, offset: %r",
+                key,
+                value,
+                sent_offset,
+            )
+        else:
+            raise RuntimeError(
```

Comment:
This will be converted to an [internal server error](https://github.com/aiven/karapace/blob/63f6b50140c3ad30183a477fc75a63d3623a33f6/karapace/rapu.py#L334-L338).

To introduce an error message the content type needs to be passed along from all the way down from the handler